coi/src/embedding.rs

```diff
@@ -37,62 +36,18 @@ where
     norm
 }
 
-/// See [`pairwise_cosine_similarity`] for details.
+/// See [`cosine_similarity`] for details.
 pub(crate) const MAXIMUM_COSINE_SIMILARITY: f32 = 1.0;
 
-/// See [`pairwise_cosine_similarity`] for details.
+/// See [`cosine_similarity`] for details.
 pub(crate) const MINIMUM_COSINE_SIMILARITY: f32 = -1.0;
 
-/// See [`pairwise_cosine_similarity`] for details.
+/// See [`cosine_similarity`] for details.
 #[cfg_attr(not(doc), allow(unreachable_pub))]
 pub const COSINE_SIMILARITY_RANGE: RangeInclusive<f32> =
     MINIMUM_COSINE_SIMILARITY..=MAXIMUM_COSINE_SIMILARITY;
 
-/// Computes the pairwise cosine similarities of vectors.
-///
-/// Each value is bounded in `[-1, 1]`. The zero vector is always "similar" to all other vectors,
-/// thus will yield a similarity of 1.
-///
-/// # Panics
-/// Panics if the vectors don't consist solely of real values or their shapes don't match.
-///
-/// Only use this with [`TrustedLen`] iterators, otherwise indexing may panic. The bound can't be
-/// named currently, because the trait is nightly-gated. [`ExactSizeIterator`] isn't a feasible and
-/// trusted replacement, for example it isn't implemented for [`Chain`]ed iterators.
-///
-/// [`TrustedLen`]: std::iter::TrustedLen
-/// [`ExactSizeIterator`]: std::iter::ExactSizeIterator
-/// [`Chain`]: std::iter::Chain
-pub fn pairwise_cosine_similarity<'a, I>(iter: I) -> Array2<f32>
-where
-    I: IntoIterator<Item = ArrayView1<'a, f32>>,
-    I::IntoIter: Clone,
-{
-    let iter = iter.into_iter();
-    let size = match iter.size_hint() {
-        (lower, Some(upper)) if lower == upper => lower,
-        (lower, None) if lower == usize::MAX => lower,
-        _ => unimplemented!("I::IntoIter: TrustedLen"),
-    };
-
-    let norms = iter.clone().map(|a| l2_norm(a.view())).collect::<Vec<_>>();
-    let mut similarities = Array2::ones((size, size));
-    for ((i, a), (j, b)) in iter
-        .clone()
-        .enumerate()
-        .cartesian_product(iter.enumerate())
-        .filter(|((i, _), (j, _))| j > i && norms[*i] > 0. && norms[*j] > 0.)
-    {
-        similarities[[i, j]] = (a.dot(&b) / norms[i] / norms[j]).clamp(-1., 1.);
-        similarities[[j, i]] = similarities[[i, j]];
-    }
-
-    similarities
-}
-
 /// Computes the cosine similarity of two vectors.
-///
-/// See [`pairwise_cosine_similarity`] for details.
```

Comment:
🔧 we need to replace this reference with the deleted paragraph from the other docs, otherwise the references at the constants above point into void.
```suggestion
/// Each value is bounded in `[-1, 1]`. The zero vector is always "similar" to all other vectors,
/// thus will yield a similarity of 1.
```

Reply:
Yes will change